src/ui/pages/SwapForm/SlippageSettings/SlippageSettings.tsx

```diff
@@ -91,6 +95,67 @@ function SlippageWarning({ percentValue }: { percentValue: string }) {
   );
 }
 
+function PercentWidth({ onValue }: { onValue(width: number): void }) {
+  const persentCharWidthRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    let interval: ReturnType<typeof setTimeout>;
+    if (persentCharWidthRef.current) {
+      onValue(persentCharWidthRef.current.clientWidth);
+    } else {
+      interval = setTimeout(
```

Comment:
Why is timeout needed here? We should be able to access dom as soon as it's mounted. We should do it even sooner than `useEffect` runs, `useLayoutEffect` is appropriate here